tortoise/src/main/scala/ProcedureCompiler.scala

```diff
@@ -18,8 +18,9 @@ class ProcedureCompiler(handlers: Handlers)(implicit compilerFlags: CompilerFlag
   def compileProcedures(procedureDefs: Seq[ProcedureDefinition]): CompiledProceduresDictionary =
     procedureDefs.map(compileProcedureDef)
 
-  private def compileProcedureDef(pd:            ProcedureDefinition)
+  private def compileProcedureDef(originalPd:    ProcedureDefinition)
                         (implicit compilerFlags: CompilerFlags, compilerContext: CompilerContext): (String, Seq[String]) = {
+    val pd = optimize.Optimizer(originalPd)
```

Comment:
That isn't taken into account here. I don't know that the Java option is viable, given that this will be running in Javascript as well as on the JVM. Perhaps we could add a new field to `CompilerFlags` and check that value here?